$OpenBSD: patch-meson_build,v 1.6 2019/06/11 11:15:04 naddy Exp $

implicit_include_directories : false
  Prevent libc++ headers from picking up <version> file; the header
  files are under include/

Index: meson.build
--- meson.build.orig
+++ meson.build
@@ -25,7 +25,7 @@ jsoncpp_gen_sources = configure_file(
   output : 'version.h',
   configuration : jsoncpp_cdata,
   install : true,
-  install_dir : join_paths(get_option('prefix'), get_option('includedir'), 'json')
+  install_dir : join_paths(get_option('prefix'), get_option('includedir'), 'jsoncpp', 'json')
 )
 
 jsoncpp_headers = [
@@ -43,8 +43,11 @@ jsoncpp_include_directories = include_directories('inc
 
 install_headers(
   jsoncpp_headers,
-  subdir : 'json')
+  subdir : 'jsoncpp/json')
 
+thread_dep = dependency('threads')
+no_asneeded = '-Wl,--no-as-needed'
+
 jsoncpp_lib = library(
   'jsoncpp',
   [ jsoncpp_gen_sources,
@@ -55,12 +58,16 @@ jsoncpp_lib = library(
     'src/lib_json/json_writer.cpp'],
   soversion : 20,
   install : true,
+  dependencies : thread_dep,
+  link_args : no_asneeded,
+  implicit_include_directories : false,
   include_directories : jsoncpp_include_directories)
 
 import('pkgconfig').generate(
   libraries : jsoncpp_lib,
   version : meson.project_version(),
   name : 'jsoncpp',
+  subdirs : 'jsoncpp',
   filebase : 'jsoncpp',
   description : 'A C++ library for interacting with JSON')
 
@@ -68,6 +75,8 @@ import('pkgconfig').generate(
 jsoncpp_dep = declare_dependency(
   include_directories : jsoncpp_include_directories,
   link_with : jsoncpp_lib,
+  dependencies : thread_dep,
+  link_args : no_asneeded,
   version : meson.project_version(),
   sources : jsoncpp_gen_sources)
 
@@ -79,8 +88,11 @@ jsoncpp_test = executable(
   [ 'src/test_lib_json/jsontest.cpp',
     'src/test_lib_json/jsontest.h',
     'src/test_lib_json/main.cpp'],
+  implicit_include_directories : false,
   include_directories : jsoncpp_include_directories,
   link_with : jsoncpp_lib,
+  dependencies : thread_dep,
+  link_args : no_asneeded,
   install : false)
 test(
   'unittest_jsoncpp_test',
@@ -89,7 +101,10 @@ test(
 jsontestrunner = executable(
   'jsontestrunner',
   'src/jsontestrunner/main.cpp',
+  implicit_include_directories : false,
   include_directories : jsoncpp_include_directories,
+  dependencies : thread_dep,
+  link_args : no_asneeded,
   link_with : jsoncpp_lib,
   install : false)
 test(
